system/mp/mpcomm.cpp

```diff
@@ -2945,46 +2949,99 @@ class CCommunicator: public ICommunicator, public CInterface
         return channel->verifyConnection(tm,true);
     }
 
-    bool verifyAll(bool duplex, unsigned timeout)
+    bool verifyAll(bool duplex, unsigned totalTimeout, unsigned perConnectionTimeout)
     {
         CriticalBlock block(verifysect);
-        CTimeMon tm(timeout);
+        CTimeMon totalTM(totalTimeout);
+
+        std::vector<std::future<bool>> threads;
+        Semaphore sem;
+        sem.signal(std::thread::hardware_concurrency());
+        bool abort = false;
```

Comment:
ok, I thought because byte it was atomic,
but I didn't realize because it's a local var, that compiler may assume it isn't changed.
Now you point out, I wonder if because it's captured and used in a lambda - the compiler is wise enough to know it might be used/changed by another thread.

Also didn't realize that making it an atomic had no overhead if only reading and writing.
I agree it is clearer as an atomic.

Will change it.